File: Medium/1887. Reduction Operations to Make the Array Elements Equal/1887. Reduction Operations to Make the Array Elements Equal.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var reductionOperations = function (nums) {
    nums.sort((a, b) => b - a); // Sort in descending order

    let operations = 0;
    let nextLargest = nums[0];

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] < nextLargest) {
            nextLargest = nums[i];
            operations += i;
        }
    }

    return operations;
};

File: Medium/1887. Reduction Operations to Make the Array Elements Equal/1887. Reduction Operations to Make the Array Elements Equal.rs
impl Solution {
    pub fn reduction_operations(nums: Vec<i32>) -> i32 {
        let mut nums = nums;
        nums.sort_unstable_by(|a, b| b.cmp(a)); // Sort in descending order
        
        let mut operations = 0;
        let mut next_largest = nums[0];
        
        for i in 1..nums.len() {
            if nums[i] < next_largest {
                next_largest = nums[i];
                operations += i as i32;
            }
        }
        
        operations
    }
}
